rust-doc-style-guide.md

```diff
@@ -0,0 +1,276 @@
+
+## Lists
+
+### Types
+
+The following types of lists are usually used in documentation:
+
+- **Bullet list** -- use it if the order or items is not important
+- **Numbered list** -- use it if the order of items is important, such as when describing a process
+  - **Procedure** -- special type of numbered list that gives steps to achieve some goal (to achieve this, do this); for an example of a procedure, see the [Usage](https://doc.rust-lang.org/nightly/rustc/profile-guided-optimization.html#usage) section in The rustc book.
+
+### Formatting
+
+The books on Rust usually use the following list formatting:
+
+- Finish an introductory sentence with a colon.
+- Capitalize the first letter of each bullet point.
+
+  > Using C or C++ inside of a Rust project consists of two major parts:
+
+  > - Wrapping the exposed C API for use with Rust
+  > - Building your C or C++ code to be integrated with the Rust code
+
+- If a bullet point is a full sentence, you can end it with a full stop.
+  - If a list has at least one full stop, end all other list items with a full stop.
+
+  > To reliably interact with these peripherals:
+
+  > - Always use `volatile` methods to read or write to peripheral memory, as it can change at any time.
+  > - In software, we should be able to share any number of read-only accesses to these peripherals.
+  > - If some software should have read-write access to a peripheral, it should hold the only reference to that peripheral.
+
+- For longer list items, consider using a summary word of phrase to make content [scannable](https://learn.microsoft.com/en-us/style-guide/scannable-content/).
+
+  > If you run Windows on your host machine, make sure ...
+  >
+  > - **MSVC**: Recommended ABI, included in ...
+  > - **GNU**: ABI used by the GCC toolchain ...
+
+  -  For an example using bold font, see the list in the [Modules Cheat Sheet](https://doc.rust-lang.org/book/ch07-02-defining-modules-to-control-scope-and-privacy.html#modules-cheat-sheet) section in The Rust Programming Language book.
+  -  For an example using monospace font, see the [Panicking](https://docs.rust-embedded.org/book/start/panicking.html#panicking) section in The Embedded Rust Book.
+
+
+## Using `monospace`
+
+Use monospace font for the following items:
+
+- Code snippets
+
+  The [Installation](https://doc.rust-lang.org/book/ch01-01-installation.html) chapter in the Rust Programming Language book suggests:
+
+  - Start the terminal commands with `$`
+  - Output of previous commands should not start with `$`
+  - For PowerShell-specific examples, use `>` instead of `$`
+
+  [book-ja](https://github.com/rust-lang-ja/book-ja/blob/master-ja/style-guide.md) also suggests:
+
+  - Use `bash` syntax highlighting
+
+- Rust declarations: commands, functions, arguments, parameters, flags, variables
+- In-line command line output
+
+  > Writing a program that prints `Hello, world!`
+
+- Data types: `i8`, `u128`
+- Names of crates, traits, libraries,
```

Comment:
Shall we also use it for container images names? Eg: `idf-rust` in https://esp-rs.github.io/book/installation/index.html#using-containers

Reply:
Added the line

- Images and containers: `idf-rust`